src/fakeredis_cluster.erl

```diff
@@ -27,10 +28,10 @@
                             EventType    :: connect | command | reply,
                             Data         :: any()}.
 
--record(state, { max_clients = 0
-               , options = []
-               , node_map = #{}
-               , slots_maps = []
+-record(state, { max_clients   = 0   :: non_neg_integer()
+               , options       = []  :: [tuple()]
+               , nodes         = #{} :: #{Id :: binary() | any() => #node{}}
```

Comment:
What is ID when it's not a binary()? Any includes everything...

Reply:
its the last line in following snippet that required it, couldnt wrap my head around it. aimed to fix it next...maybe you see why?

```
Error: The call maps:get(Id::{binary(),char()}, #{binary()=>#node{id::binary(),address::binary(),port::integer()}}) will never return since the success typing arguments are (any(),map())

handle_call({get_redirect_by_key, Key}, _From, State) ->
    Slot = fakeredis_hash:hash(Key),
    case State#state.ask_redirects of
        #{Key := Id} ->
            %% ASK redirect exists for this key
            #node{address = Addr,
                  port    = Port} = maps:get(Id, State#state.nodes),

```

Reply:
found an error in the #state.ask_redirects spec which surfaced here, corrected now

Reply:
`Id` here is `{Host, Port}` (as can be seen in the error message), so it should not be looked up in the State#state.nodes.

Reply:
The problem was that the

Reply:
Oh, you're right, it's actually a node id. I see now in `set_ask_redirect` that the host and port passed by the user is looked up to make sure the redirect is for an existing node, and then a node id is saved instead of host and port, so you're right changing the type of `#state.ask_redirects`. :-) Good!